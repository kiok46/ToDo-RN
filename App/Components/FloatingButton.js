import React, { Component } from 'react';
import {
    Text,
    View,
    StyleSheet,
    TouchableOpacity
} from 'react-native';
import PropTypes from 'prop-types';
import { Entypo } from '@expo/vector-icons';

class FloatingButton extends Component {
	constructor(props) {
		super(props);
	}
	render (){
		return (
         <TouchableOpacity onPress={this.props.onPress}>
            <View style={styles.floatingStyle}>
                <Entypo name="plus" size={32} color="white" />
            </View>
         </TouchableOpacity>
		);
	}
};

const styles = StyleSheet.create({
    floatingStyle: {
        position: 'absolute',
        bottom: 30,
        right: 30,
        height: 60,
        width: 60,
        backgroundColor: 'tomato',
        borderRadius: 30,
        justifyContent: 'center',
        alignItems: 'center'
    },
});


FloatingButton.defultProps = {
    onPress: () => {}
}

FloatingButton.propTypes = {
    onPress: PropTypes.func
}

export default FloatingButton;