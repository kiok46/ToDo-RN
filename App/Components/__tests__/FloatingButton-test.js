import React from 'react';
import FloatingButton from '../FloatingButton';
import { shallow } from 'enzyme';


describe('testing FloatingButton rendering', () => {
    it('renders as expected', () => {
        const wrapper = shallow(
            <FloatingButton />
        )
        expect(wrapper).toMatchSnapshot();
    })
})
