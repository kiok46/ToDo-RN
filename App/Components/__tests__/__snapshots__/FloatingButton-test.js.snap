// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing FloatingButton rendering renders as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FloatingButton />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "activeOpacity": 0.2,
      "children": <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "tomato",
            "borderRadius": 30,
            "bottom": 30,
            "height": 60,
            "justifyContent": "center",
            "position": "absolute",
            "right": 30,
            "width": 60,
          }
        }
      >
        <Icon
          allowFontScaling={false}
          color="white"
          name="plus"
          size={32}
        />
      </View>,
      "onPress": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Icon
          allowFontScaling={false}
          color="white"
          name="plus"
          size={32}
        />,
        "style": Object {
          "alignItems": "center",
          "backgroundColor": "tomato",
          "borderRadius": 30,
          "bottom": 30,
          "height": 60,
          "justifyContent": "center",
          "position": "absolute",
          "right": 30,
          "width": 60,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "allowFontScaling": false,
          "color": "white",
          "name": "plus",
          "size": 32,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeOpacity": 0.2,
        "children": <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "tomato",
              "borderRadius": 30,
              "bottom": 30,
              "height": 60,
              "justifyContent": "center",
              "position": "absolute",
              "right": 30,
              "width": 60,
            }
          }
        >
          <Icon
            allowFontScaling={false}
            color="white"
            name="plus"
            size={32}
          />
        </View>,
        "onPress": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Icon
            allowFontScaling={false}
            color="white"
            name="plus"
            size={32}
          />,
          "style": Object {
            "alignItems": "center",
            "backgroundColor": "tomato",
            "borderRadius": 30,
            "bottom": 30,
            "height": 60,
            "justifyContent": "center",
            "position": "absolute",
            "right": 30,
            "width": 60,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "allowFontScaling": false,
            "color": "white",
            "name": "plus",
            "size": 32,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
